### PR TITLE
Make log4j and logback appender configurable

### PR DESCRIPTION
**Pull request by heyams**
Fix the duplicate in https://github.com/Azure-Samples/ApplicationInsights-Java-Samples/pull/98

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java [hunk 1]
```diff
@@ -150,10 +169,20 @@ private static void enableInstrumentations(
...
-    properties.put("otel.instrumentation.log4j-appender.enabled", "true");
-    if (otelConfig.getBoolean("otel.instrumentation.logback-appender.enabled", true)) {
-      properties.put("otel.instrumentation.logback-appender.enabled", "true");
+    if (config.instrumentation.log4jAppender.enabled) {
+      properties.put("otel.instrumentation.log4j-appender.enabled", "true");
+    } else {
+      properties.put("otel.instrumentation.log4j-appender.enabled", "false");
+    }
+    if (config.instrumentation.logbackAppender.enabled) {
+      if (otelConfig.getBoolean("otel.instrumentation.logback-appender.enabled", true)) {
```
Comment (jeanbisutti) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:178]: Could you explain the need for `otelConfig.getBoolean("otel.instrumentation.logback-appender.enabled", true)`?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java [hunk 2]
```diff
@@ -53,22 +53,41 @@ public Map<String, String> apply(ConfigProperties otelConfig) {
         configuration.preview.captureHttpClientHeaders.responseHeaders);
 
     // enable capturing all mdc properties
```
Comment (jeanbisutti) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:55]: Coud you gather together all the `if(configuration.instrumentation.log4jAppender.enabled)`?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java [hunk 3]
```diff
@@ -53,22 +53,41 @@ public Map<String, String> apply(ConfigProperties otelConfig) {
...
+      properties.put(
+          "otel.instrumentation.log4j-appender.experimental.capture-map-message-attributes",
+          "true");
+    } else {
+      properties.put(
+          "otel.instrumentation.log4j-appender.experimental.capture-map-message-attributes",
+          "false");
+    }
 
+    if (configuration.instrumentation.logbackAppender.enabled) {
```
Comment (jeanbisutti) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:70]: Coould you gather together all the `if(configuration.instrumentation.logbackAppender.enabled)`?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java (hunk 1 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:178]: i don't know the default value upstream. It's better to be explicit.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java (hunk 2 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:55]: it's grouped based on comments.

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java (hunk 3 above)
Comment (heyams) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AiConfigCustomizer.java:70]: same here

**Pull request closed without merging**